File: src/main/java/com/lessmarkup/dataobjects/UserBlockHistory.java
package com.lessmarkup.dataobjects;

import com.lessmarkup.interfaces.data.AbstractDataObject;
import com.lessmarkup.interfaces.recordmodel.RequiredField;

import java.time.OffsetDateTime;

public class UserBlockHistory extends AbstractDataObject {
    private long userId;
    private long blockedByUserId;
    @RequiredField
    private OffsetDateTime created;
    private OffsetDateTime blockedToTime;
    private boolean unblocked;
    @RequiredField
    private String reason;
    @RequiredField
    private String internalReason;

    public long getUserId() {
        return userId;
    }

    public void setUserId(long userId) {
        this.userId = userId;
    }

    public long getBlockedByUserId() {
        return blockedByUserId;
    }

    public void setBlockedByUserId(long blockedByUserId) {
        this.blockedByUserId = blockedByUserId;
    }

    public OffsetDateTime getCreated() {
        return created;
    }

    public void setCreated(OffsetDateTime created) {
        this.created = created;
    }

    public OffsetDateTime getBlockedToTime() {
        return blockedToTime;
    }

    public void setBlockedToTime(OffsetDateTime blockedToTime) {
        this.blockedToTime = blockedToTime;
    }

    public boolean isUnblocked() {
        return unblocked;
    }

    public void setUnblocked(boolean unblocked) {
        this.unblocked = unblocked;
    }

    public String getReason() {
        return reason;
    }

    public void setReason(String reason) {
        this.reason = reason;
    }

    public String getInternalReason() {
        return internalReason;
    }

    public void setInternalReason(String internalReason) {
        this.internalReason = internalReason;
    }
}

File: src/main/scala/com/lessmarkup/dataobjects/UserBlockHistory.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.dataobjects

import java.time.OffsetDateTime

import com.lessmarkup.interfaces.annotations.RequiredField
import com.lessmarkup.interfaces.data.DataObject

class UserBlockHistory extends DataObject {
  var userId: Long = 0
  var blockedByUserId: Long = 0
  @RequiredField
  var created: OffsetDateTime = null
  var blockedToTime: OffsetDateTime = null
  var unblocked: Boolean = false
  @RequiredField
  var reason: String = null
  @RequiredField
  var internalReason: String = null
}
